torch/_inductor/mkldnn_lowerings.py

```diff
@@ -544,24 +547,79 @@ def qlinear_unary(
             bias_dtype = None if bias is None else bias.get_dtype()
 
             choices: List[ChoiceCaller] = []
+
             if use_max_autotune():
+                # NOTE: since vector weight scale is supported with oneDNN qlinear in PyTorch,
+                # we are able to support the case of per-token quantized activation & per-channel quantized weights
+                # if x_zp is not present in a pattern, since in that case, we can apply the actual scale of x
+                # after oneDNN qlinear in a replacement pattern. This is being done for a smoothquant pattern.
+                # Inductor codegen is able to make it an epilogue fusion.
+                # We don't support the case of per-token quantized activation with oneDNN qlinear if
+                # the activation is quantized asymmetrically, or if the per-token quantized activation is
+                # quantized symmetrically but still has a corresponding zero-points tensor,
+                # which would be all zeros in case of int8).
+
                 *_, layout, x, packed_weight = mm_args(
                     x, packed_weight, layout=layout, out_dtype=output_dtype
                 )
+
+                def _is_sym_quantized_wgt():
+                    # weight zero point may not be a ConstantBuffer, but a computeBuffer instead.
+                    # That may happen if it's dummy.
+                    # In some Inductor pattern, we may insert a dummy weight zero-point even if it doesn't exist
+                    # just because oneDNN qlinear has a w_zp parameter.
```

Comment:
I don't think we need to mention the implementation details here.

Reply:
But how would someone new to the code understand why we can encounter a ComputeBuffer for a dummy w_zp here, which is created during replacing a SmoothQuant Inductor pattern?